Index: src/test/java/tests/BaseTest.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.BaseRevisionTextPatchEP
<+>package tests;\r\n\r\nimport aquality.selenium.browser.AqualityServices;\r\nimport org.testng.annotations.AfterMethod;\r\nimport org.testng.annotations.BeforeMethod;\r\nimport pages.MainPage;\r\nimport utils.SettingsTestData;\r\n\r\nimport static aquality.selenium.browser.AqualityServices.getBrowser;\r\n\r\npublic abstract class BaseTest {\r\n    protected final MainPage mainPage = new MainPage();\r\n\r\n    @BeforeMethod\r\n    public void setup() {\r\n        getBrowser().maximize();\r\n        getBrowser().goTo(SettingsTestData.getEnvData().getHost());\r\n    }\r\n\r\n    @AfterMethod\r\n    public void teardown() {\r\n        if (AqualityServices.isBrowserStarted()) {\r\n            getBrowser().quit();\r\n        }\r\n    }\r\n}\r\n
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/test/java/tests/BaseTest.java b/src/test/java/tests/BaseTest.java
--- a/src/test/java/tests/BaseTest.java	(revision 47cd5f5f1e863790eb6f9bb63965f38d8fd376bd)
+++ b/src/test/java/tests/BaseTest.java	(date 1727607783558)
@@ -13,8 +13,8 @@
 
     @BeforeMethod
     public void setup() {
-        getBrowser().maximize();
-        getBrowser().goTo(SettingsTestData.getEnvData().getHost());
+       getBrowser().maximize();
+       getBrowser().goTo(SettingsTestData.getEnvData().getHost());
     }
 
     @AfterMethod
Index: src/test/resources/environment/prod.json
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.BaseRevisionTextPatchEP
<+>{\r\n  \"wait\": 10,\r\n  \"host\": \"https://www.wikipedia.org/\"\r\n}\r\n
===================================================================
diff --git a/src/test/resources/environment/prod.json b/src/test/resources/environment/prod.json
--- a/src/test/resources/environment/prod.json	(revision 47cd5f5f1e863790eb6f9bb63965f38d8fd376bd)
+++ b/src/test/resources/environment/prod.json	(date 1727606866099)
@@ -1,4 +1,4 @@
 {
   "wait": 10,
-  "host": "https://www.wikipedia.org/"
+  "host": "https://www.accuweather.com/"
 }
Index: src/test/java/pages/MainPage.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.BaseRevisionTextPatchEP
<+>package pages;\r\n\r\nimport aquality.selenium.elements.interfaces.IButton;\r\nimport aquality.selenium.elements.interfaces.ITextBox;\r\nimport aquality.selenium.forms.Form;\r\nimport org.openqa.selenium.By;\r\n\r\npublic class MainPage extends Form {\r\n    private final ITextBox searchField = getElementFactory().getTextBox(By.id(\"searchInput\"), \"Search field\");\r\n    private final IButton submitButton = getElementFactory().getButton(By.xpath(\"//form[@id='search-form']//button\"), \"Submit button\");\r\n\r\n    public MainPage() {\r\n        super(By.xpath(\"//h1[@class='central-textlogo-wrapper']//span\"), \"Wikipedia\");\r\n    }\r\n\r\n    public void searchForText(String text) {\r\n        searchField.clearAndType(text);\r\n        submitButton.click();\r\n    }\r\n}\r\n
===================================================================
diff --git a/src/test/java/pages/MainPage.java b/src/test/java/pages/MainPage.java
--- a/src/test/java/pages/MainPage.java	(revision 47cd5f5f1e863790eb6f9bb63965f38d8fd376bd)
+++ b/src/test/java/pages/MainPage.java	(date 1727608097719)
@@ -1,20 +1,32 @@
 package pages;
 
+import aquality.selenium.elements.ElementType;
 import aquality.selenium.elements.interfaces.IButton;
+import aquality.selenium.elements.interfaces.ILabel;
 import aquality.selenium.elements.interfaces.ITextBox;
 import aquality.selenium.forms.Form;
 import org.openqa.selenium.By;
 
+import java.util.List;
+
 public class MainPage extends Form {
-    private final ITextBox searchField = getElementFactory().getTextBox(By.id("searchInput"), "Search field");
-    private final IButton submitButton = getElementFactory().getButton(By.xpath("//form[@id='search-form']//button"), "Submit button");
+    private final ITextBox searchBar = getElementFactory().getTextBox(By.className("search-input"),"Search Input");
+    private final ILabel resultsContainer = getElementFactory().getLabel(By.className("results-container"),"Results container");
+    private final List<ILabel> searchBarResults = getElementFactory().findElements(By.className("search-bar-result"), ElementType.LABEL);
 
-    public MainPage() {
-        super(By.xpath("//h1[@class='central-textlogo-wrapper']//span"), "Wikipedia");
+    public MainPage(){
+        super(By.xpath("//div[@class='featured-search-bar']"),"Main page");
+    }
+
+    public void searchInput(String searchInput) {
+        searchBar.clearAndType(searchInput);
+    }
+
+    public boolean resultsListDisplayed() {
+        return resultsContainer.state().waitForDisplayed();
     }
 
-    public void searchForText(String text) {
-        searchField.clearAndType(text);
-        submitButton.click();
+    public void clickFirstResult() {
+        searchBarResults.get(1).click();
     }
 }
